src/controllers/user.controller.ts

```diff
@@ -111,3 +111,62 @@ export const signUpUser: RequestHandler = async (req, res) => {
   }
 };
 
+export const forgotPassword: RequestHandler = async (req, res)=>{
+  try {
+    const {email} = req.body
+
+    //validate email for existence
+    const checkEmail = await User.findOne({ where: { email } });
+    if (!checkEmail) {
+      return res.status(404).json({
+        message: "Email not found"
+      })
+    }
+
+    // generate password token
+    const passwordToken = jwt.sign({
+      userId : checkEmail.userId,
+      userName : checkEmail.userName
+    }, <string>process.env.JWT_SECRET_TOKEN, {
+      expiresIn: "5m"
+    })
+
+
+    //send the password resest link to the user email address
+
+    const emailContent: Content = {
+      body: {
+        name: email,
+        intro: ` Welcome to Social-commerce! Please click on the link to reset your password:`,
+        action: {
+          instructions: `Here's the link to reset your password below:`,
+          button: {
+            color: '#673ee6',
+            text: "Note: This link will expire in 5(five) minutes",
+            link: "#",
```

Comment:
this link is suppose to send the route to the change password link boss!

Reply:
Done boss,  remaining to add zod validators